keps/sig-storage/3107-csi-nodeexpandsecret/README.md

```diff
@@ -220,53 +237,137 @@ N/A
 
 ### Rollout, Upgrade and Rollback Planning
 
-TBD
-
 ###### How can a rollout or rollback fail? Can it impact already running workloads?
 
-TBD
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
 
-###### What specific metrics should inform a rollback?
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
 
-TBD
+###### What specific metrics should inform a rollback?
 
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-TBD
-
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
 
-TBD
-
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
 ### Monitoring Requirements
 
-TBD
-
 ###### How can an operator determine if the feature is in use by workloads?
 
-TBD
-
-###### How can someone using this feature know that it is working for their instance?
+An operator can query for api server and kubelet flags in the cluster
+for `CSINodeExpandSecret` flag and if it exist then the feature is
+in use.
+
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason:
+- [ ] API .status
+  - Condition name:
+  - Other field:
+- [x] Other (treat as last resort)
+  - Details: to make use of this feature in a cluster a StorageClass instance  has
+to carry below entries in the parameter list.
+
+  ```
+  csi.storage.k8s.io/node-expand-secret-name
+  csi.storage.k8s.io/node-expand-secret-namespace
+ ```
+
+  The subjected CSI PV object should have `nodeExpandSecretRef` field filled with the
+  details given in the StorageClass.
```

Comment:
Does the presence of `nodeExpandSecretRef` indicate that the feature is "in use" or that it is "working properly", or both? If it is the former, move up to "How can an operator determine if the feature is in use by workloads?" section?

Reply:
It is both. specifically considering the SC belong to specific provisioner